src/backends/beamcoder.ts

```diff
@@ -399,8 +402,20 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
             this.#packetReadCount++;
         }
 
+        // we read through all the available packets and frames, but we still don't have a frame. This can happen
+        // when our targetPTS is to close to the end of the video. In this case, we'll try to seek further away from
+        // the end of the video and try again. We've set up a MAX_RECURSION to prevent an infinite loop.
         if (!outputFrame) {
-            throw Error('No matching frame found');
+            if (MAX_RECURSION < this.#recursiveReadCount) {
+                throw Error('No matching frame found');
+            }
+            const TIME_OFFSET = 0.1; // time offset in seconds
+            const PTSOffset = this._timeToPTS(TIME_OFFSET);
+            this.#recursiveReadCount++;
+            outputFrame = await this._getFrameAtPts(targetPTS, SeekPTSOffset + PTSOffset);
```

Comment:
Should this be a `-`?

Reply:
I don't understand why it would work better the next time if we seek further @animanathome

Reply:
Because we'll have flushed/processed more packets? Shouldn't we explicitly parse the packets/flush here then?

Reply:
Nope, + because we're subtracting it afterwards. See https://github.com/Lumen5/framefusion/blob/3d85cc6e7b721367d83e4b985923c13ae3b6fe9f/src/backends/beamcoder.ts#L312

Reply:
Ahhh ok, I think it would be clearer if the put the `-` here and the `+` at line 312.

Reply:
Hmmm, `targetPTS - SeekPTSOffset` says that we're removing something or moving backwards, which is what we do. If we do `targetPTS + SeekPTSOffset` I would read it more as we're adding an offset or moving forward, which we don't do.

Reply:
Currently, we go backward, but this param is exposed publicly and could be use to seek to add a positive offset? It would not make so much sense then to have it be negative to go forward

Reply:
Right, I didn't think about it like that. Also, `_getFrameAtPts` is not expected to be used publicly.